lib/png/png.go

```diff
@@ -0,0 +1,103 @@
+package png
+
+import (
+	"bytes"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	_ "embed"
+
+	"github.com/playwright-community/playwright-go"
+)
+
+func InitPlaywright() (*playwright.Playwright, playwright.Browser, error) {
+	// check if playwright driver/browsers are installed and up to date
+	// https://github.com/playwright-community/playwright-go/blob/8e8f670b5fa7ba5365ae4bfc123fea4aac359763/run.go#L64.
+	driver, err := playwright.NewDriver(&playwright.RunOptions{})
+	if err != nil {
+		return nil, nil, err
+	}
+	if _, err := os.Stat(driver.DriverBinaryLocation); errors.Is(err, os.ErrNotExist) {
+		err = playwright.Install()
+		if err != nil {
+			return nil, nil, err
+		}
+	} else if err == nil {
+		cmd := exec.Command(driver.DriverBinaryLocation, "--version")
+		output, err := cmd.Output()
+		if err != nil || !bytes.Contains(output, []byte(driver.Version)) {
+			err = playwright.Install()
+			if err != nil {
+				return nil, nil, err
+			}
+		}
+	}
+
+	pw, err := playwright.Run()
+	if err != nil {
+		return nil, nil, err
+	}
+	browser, err := pw.Chromium.Launch()
+	if err != nil {
+		return nil, nil, err
+	}
+	return pw, browser, nil
+}
+
+//go:embed generate_png.js
+var genPNGScript string
+
+func ExportPNG(browser playwright.Browser, svg []byte) (outputImage []byte, err error) {
+	var page playwright.Page
+	defer func() error {
+		err = page.Close()
+		if err != nil {
+			return err
+		}
+		return nil
+	}()
+
+	if browser == nil {
+		return nil, fmt.Errorf("browser is not initialized for png export")
+	}
+	page, err = browser.NewPage()
+	if err != nil {
+		return nil, err
+	}
+	encodedSVG := base64.StdEncoding.EncodeToString(svg)
+	pngInterface, err := page.Evaluate(genPNGScript, "data:image/svg+xml;charset=utf-8;base64,"+encodedSVG)
```

Comment:
looking at the API for Evaluate, it looks like u also have to check for values like `undefined`, which might have err = nil but were unserializable. there are others, look at their docs or source code for exhaustive list. 

or maybe u can have the javascript return an object, with single field "data", and unmarshal it here. if the unmarshaling fails, it means something went wrong